lnrpc/routerrpc/router.proto

```diff
@@ -125,8 +124,27 @@ message PaymentStatus {
     The taken route when state is SUCCEEDED.
     */
     lnrpc.Route route = 3;
+
+    /**
+    List of all shard statuses to this payment_hash [EXPERIMENTAL].
+    */
+    repeated ShardStatus shards = 4 [json_name = "shards"];
 }
 
+message ShardStatus {
+    /// Current state the shard is in.
+    PaymentState state = 1;
+
+    /**
+    The preimage of the shard when state is SUCCEEDED.
+    */
+    bytes preimage = 2;
+
+    /**
+    The taken route when state is SUCCEEDED.
```

Comment:
This is actually no longer correct. Also for failed payments, we report the route. This is useful for probe payments to find out the final route taken.